test_conformance/c11_atomics/test_atomics.cpp

```diff
@@ -1615,6 +1615,30 @@ class CBasicTestFlag : public CBasicTestMemOrderScope<HostAtomicType, HostDataTy
       orderStr = std::string(", ") + get_memory_order_type_name(MemoryOrderForClear());
     return orderStr + MemoryScopeStr();
   }
+
+  virtual int ExecuteSingleTest(cl_device_id deviceID, cl_context context,
+                                cl_command_queue queue)
+  {
+      // This test assumes support for the memory_scope_device scope in the case
+      // that LocalMemory() == false. Therefore we should skip this test in that
+      // configuration on a 3.0 driver since supporting the memory_scope_device
+      // scope is optionaly.
+      if (get_device_cl_version(deviceID) >= Version{ 3, 0 })
+      {
+          if (!LocalMemory()
+              && !(gAtomicMemCap & CL_DEVICE_ATOMIC_SCOPE_DEVICE))
+          {
+              log_info(
+                  "Skipping atomic_flag test due to use of atomic_scope_device "
+                  "which is optionally not supported on this device\n");
+              return TEST_SKIPPED_ITSELF;
```

Comment:
All the other `ExecuteSingleTest` have code like this:

```c
    if (CheckCapabilities(MemoryScope(), MemoryOrder()) == TEST_SKIPPED_ITSELF)
        return 0; // skip test - not applicable
```

This has two differences to the code here.  The use of `CheckCapabilities(...)`, and `return 0;`.

Does the dependence on `!LocalMemory()` prevent use of `CheckCapabilities(...)`?

As for the return code, I'm not sure if `TEST_SKIPPED_ITSELF` is valid here but `0` would be more consistent with the rest of the tests.

Reply:
Thanks for the review! So I'm not 100% certain, but I think that `MemoryScope()` and `MemoryOrder()` are independent of `LocalMemory()`. For a device that doesn't support `memory_scope_device` `MemoryScope()` shouldn't ever return `memory_scope_device` (after #944 is merged). But I think `LocalMemory()` is getting set separately and this is used to select the scope in the kernel source, so this is what we need to use in this case to optionally skip this test?

Reply:
Updated to return `0` to be consistent with other tests.